### PR TITLE
Fixed bug with build effects

### PR DESCRIPTION
**Pull request by Crotalus**
Paused commanders still had the build effects going.

~~Did some refactoring so the build effect stuff is handled in the base Unit class instead.~~

### CODE REVIEW

**Comment by Crotalus** [comment]
An excercise for the future could be to merge self.UnitBeingBuilt and self.BuildingUnit also, could as well check if self.UnitBeingBuilt is nil

**Comment by ChrisKitching** [comment]
We should not be moving specialised logic into Unit.

We really need to introduce classes for "things that build other things", but that's not something we should do on release day.
Instead, for now, let's do the minimum required to fix this bug, duplicated as necessary, and cull it later on when we make the inheritance chains for construction units vaguely sensible.

**Comment by Crotalus** [comment]
There, changed the patch into a quickfix instead

**Comment by ChrisKitching** [comment]
Oh,that was quick, thanks :D

If you'd like to lend a hand on the overall unfucking of all this crap, take a look at:
https://github.com/FAForever/fa/issues/647

(It's more of a problem statement than anything terribly helpful: I've not really made much of a start).
There's a bunch of duplicated logic between the various things that build other things, and tons of stuff lurking in Unit. Really, we want to introduce base classes for stuff-that-builds-stuff and pull all the specialised crap out of Unit. Unit is already way too big (but recently we've made some progress pulling specialised stuff out of it, such as all the air transport rubbish).

File: lua/defaultunits.lua [hunk 1]
```diff
@@ -1749,12 +1749,11 @@ ConstructionUnit = Class(MobileUnit) {
...
-    if unitBeingBuilt.WorkItem.Slot and unitBeingBuilt.WorkProgress == 0 then
-        return
-    else
-        MobileUnit.OnStartBuild(self,unitBeingBuilt, order)
-    end
+        if unitBeingBuilt.WorkItem.Slot and unitBeingBuilt.WorkProgress == 0 then
+            return
+        else
+            MobileUnit.OnStartBuild(self,unitBeingBuilt, order)
+        end
```
Comment (ChrisKitching) [inline lua/defaultunits.lua]: Am I going blind, or is this actually a no-op chunk?

Comment (Crotalus) [inline lua/defaultunits.lua]: Yeah, indenting

**Pull request merged**